bears/c_languages/GNUIndentBear.py

```diff
@@ -23,18 +23,166 @@ class GNUIndentBear:
+            Uses GNU coding style.
+        :param k_and_r_style:
+            Uses Kernighan & Ritchie coding style.
+        :param linux_style:
+            Uses Linux coding style.
+        :param tab_width:
+            Number of spaces per indent level.
+        :param indent_cli_options:
+            Any command line options the indent binary understands. They
+            will be simply passed through.
         """
-        args = ("--no-tabs" if use_spaces else "--use-tabs",
-                "--line-length", str(max_line_length),
-                "--indent-level", str(tab_width),
-                "--tab-size", str(tab_width))
-        return args + tuple(shlex.split(indent_cli_options))
+        indent_options = ("--no-tabs" if use_spaces else "--use-tabs",
+                          "--line-length", str(max_line_length),
+                          "--indent-level", str(tab_width),
+                          "--tab-size", str(tab_width), )
+        indent_options += (("--cuddle-do-while",)
+                           if while_and_brace_on_same_line
+                           else ("--dont-cuddle-do-while",))
+        indent_options += (("--swallow-optional-blank-lines",)
+                           if delete_optional_blank_lines else ("-nsob",))
+        indent_options += (("--blank-lines-after-declarations",)
+                           if blank_lines_after_declarations else ("-nbad",))
+        indent_options += (("--blank-lines-after-commas",)
+                           if blank_lines_after_commas else ("-nbc",))
+        indent_options += (("--blank-lines-after-procedures",)
+                           if blank_lines_after_procedures else ("-nbap",))
+        indent_options += (("-di" + str(declaration_indent),)
+                           if declaration_indent != 0 else ())
+        indent_options += (("--case-indentation"+str(case_indentation),)
+                           if case_indentation != 0 else ())
+        indent_options += (("--space-special-semicolon",)
+                           if space_before_semicolon_after_empty_loop
+                           else ("-nss",))
+        indent_options += ("--brace-indent"+str(brace_indent),)
+        indent_options += (("--braces-on-func-def-line",)
+                           if braces_on_func_def_line else ("-blf",))
+        indent_options += ((("-ce",) if cuddle_else else ("-nce",)) +
+                           ("-br",)) if braces_on_if_line else ("-bl",)
+        indent_style_option = ()
+        indent_style_option += ("--gnu-style",) if gnu_style else ()
+        indent_style_option += (("--k-and-r-style",)
+                                if k_and_r_style and indent_style_option is ()
+                                else ())
+        indent_style_option += (("--linux-style",)
+                                if linux_style and indent_style_option is ()
+                                else ())
+        return (indent_style_option if indent_style_option is not ()
```

Comment:
I added a line before the return statement is that enough :)

2016-06-17 13:06 GMT+01:00 Abhay Raizada <notifications@github.com>:

> In bears/c_languages/GNUIndentBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/519#discussion_r67500013>
> :
>
> > +                           if space_before_semicolon_after_empty_loop
> > +                           else ("-nss",))
> > +        indent_options += ("--brace-indent"+str(brace_indent),)
> > +        indent_options += (("--braces-on-func-def-line",)
> > +                           if braces_on_func_def_line else ("-blf",))
> > +        indent_options += ((("-ce",) if cuddle_else else ("-nce",)) +
> > +                           ("-br",)) if braces_on_if_line else ("-bl",)
> > +        indent_style_option = ()
> > +        indent_style_option += ("--gnu-style",) if gnu_style else ()
> > +        indent_style_option += (("--k-and-r-style",)
> > +                                if k_and_r_style and indent_style_option is ()
> > +                                else ())
> > +        indent_style_option += (("--linux-style",)
> > +                                if linux_style and indent_style_option is ()
> > +                                else ())
> > +        return (indent_style_option if indent_style_option is not ()
>
> then please add that in the description?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/519/files/782693b61e840fb9fe264a36af6ec348ad71b29b#r67500013>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJ9xolNIX2lKPQkDDimmoifcWx6lS5aSks5qMo3NgaJpZM4I02l0>
> .
>